src/services/restaurant.services.js

```diff
@@ -1,42 +1,135 @@
 import { Restaurant } from "../models/restScheme.js"
-import {toTitleCase} from "../utils/util.js"
 
-const getRestaurantById = async (id)=> {
-    return await Restaurant.findById(id);
+// const getRestaurantById = async (id)=> {
+//     return await Restaurant.findById(id);
+// }
+
+// const getAllRestaurant = async ()=> {
+//     return await Restaurant.find();
+// }
+// const getRestaurantByName = async (name)=> {
+//     return await Restaurant.findOne({name}).populate('category.dishes');
+// }
+
+// const addRestaurant = async (restaurant) => {
+//     try {
+//         const newRest = await restaurant.save();   
+//     } catch (error) {
+//         console.log(error);
+//     }
+// }
+
+// const removeRestaurant = async (name)=> {
+//     await Restaurant.deleteOne({name});
+// }
+
+// const updateRestaurant = async (name, newName=name, desc, street, city, zip, phone, image) =>{
+//     const address = {street, city, zip};
+//     const openingHours = [{day: "a", hours: "1200-1600"},{day: "b", hours: "1200-1600"}]
+//     const category = [{title: "category", subtitle: "category"}]
+//     await Restaurant.findOneAndUpdate({name}, {name:newName, 
+//         description: desc, 
+//         address, 
+//         phone, 
+//         image, 
+//         category, 
+//         openingHours})
+// }
+
+// const RestaurantApi = {getAllRestaurant, getRestaurantById, getRestaurantByName, addRestaurant, removeRestaurant, updateRestaurant};
+
+
+/**
+ * 
+ * @param {*} Parameters | restaurant object
+ * @returns | return new restaurant
+ */
+const add = async (restaurant) => {
+    try{
+        const newRestaurant = await restaurant.save();
+        if(newRestaurant){
+            return newRestaurant;
+        }
+    }catch (error) {
+        console.log(error);
+    }
 }
 
-const getAllRestaurant = async ()=> {
-    return await Restaurant.find();
+/**
+ * 
+ * @param {*} Parameters | restaurant id
+ * @returns | return restaurant
+ */
```

Comment:
@param {*} id type String | the id of the restaurant that we want get from database
@returns type Restaurant | the restaurant from the database